src/parsley/utils.js

```diff
@@ -2,29 +2,37 @@ define('parsley/utils', function () {
   return {
     // Parsley DOM-API
     // returns object from dom attributes and values
-    // if attr is given, returns bool if attr present in DOM or not
-    attr: function ($element, namespace, checkAttr) {
+    attr: function ($element, namespace, obj) {
       var
         attribute,
-        obj = {},
         msie = this.msieversion(),
         regex = new RegExp('^' + namespace, 'i');
 
+      if ('undefined' === typeof obj)
+        obj = {};
+      else {
+        // Clear all existing values. This won't affect prototype's values
```

Comment:
Right, prototype's values are not affected by `delete`, but just for clarity it's probably better to add it. Done.